UnityProject/Assets/Scripts/Electricity/PowerSupplies/SMES.cs

```diff
@@ -36,19 +34,30 @@ public override void OnStartClient()
 		UpdateState(isOn);
 	}
 
-	public override bool Interact(GameObject originator, Vector3 position, string hand)
+    public InteractionResult Interact(HandApply interaction)
 	{
-		if (!isServer)
-		{
-			InteractMessage.Send(gameObject, hand);
+		// We can still re-use validators if we want:
+		ValidationResult validationResult = CanApply.ONLY_IF_CONSCIOUS.Validate(interaction, NetworkSide.CLIENT);
```

Comment:
maybe @chairbender could look at this and comment if IF2 system has been used properly here

Reply:
yeah, I found it a bit hard understanding it so hopefully it will be okay